templates/components/vectordbs/python/none/generate.py

```diff
@@ -16,15 +16,17 @@
 
 
 def generate_datasource():
-    logger.info("Creating new index")
-    # load the documents and create the index
-    documents = get_documents()
-    index = VectorStoreIndex.from_documents(
-        documents,
-    )
-    # store it for later
-    index.storage_context.persist(STORAGE_DIR)
-    logger.info(f"Finished creating new index. Stored in {STORAGE_DIR}")
+    storage_context = StorageContext.from_defaults()
+
+    docs = []
+
+    for doc in get_documents():
```

Comment:
why not returning all documents here (instead of the iterator)?

Reply:
ah, we may have more actions for each returned document so i implemented the iterator here, will convert to the document list if the iterator is not needed.